crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -28,6 +32,160 @@ use super::{
     TypeQualifiers, TypeVarInstance,
 };
 
+/// Tries to find declarations/bindings of an instance attribute named `name` that are only
+/// "implicitly" defined in a method of the class that corresponds to `class_body_scope`.
+#[allow(clippy::needless_pass_by_value)]
+#[salsa::tracked]
+fn implicit_instance_attribute<'db>(
```

Comment:
What's the reason for moving this function out of `Class`? It makes the diff harder to review